salt/modules/network.py

```diff
@@ -522,103 +522,109 @@ def _ip_route_linux():
                     "netmask": address_mask["netmask"],
-                    "flags": "U",
-                    "interface": ip_interface,
+                    "flags": flags,
+                    "interface": ip_interface if ip_interface else "",
                 }
             )
 
     # table all closest to old netstat inet6 output
     cmd = "ip -6 route show table all"
     out = __salt__["cmd.run"](cmd, python_shell=True)
     for line in out.splitlines():
-        comps = line.split()
-
+        line = p.search(line)
+        if line is None:
+            continue
         # need to fake similar output to that provided by netstat
         # to maintain output format
-        if comps[0] == "unreachable":
+        if line.group("unreachable") == "unreachable":
             continue
 
-        if comps[0] == "default":
-            ip_interface = ""
-            if comps[3] == "dev":
-                ip_interface = comps[4]
+        if line.group("network") == "default":
+            ip_interface = line.group("interface")
+            ip_gateway = line.group("gateway")
 
             ret.append(
                 {
                     "addr_family": "inet6",
                     "destination": "::/0",
-                    "gateway": comps[2],
+                    "gateway": ip_gateway,
                     "netmask": "",
                     "flags": "UG",
-                    "interface": ip_interface,
+                    "interface": ip_interface if ip_interface else "",
                 }
             )
 
-        elif comps[0] == "local":
-            ip_interface = ""
-            if comps[2] == "dev":
-                ip_interface = comps[3]
-
-            local_address = comps[1] + "/128"
+        elif line.group("network") == "local":
+            ip_interface = line.group("interface")
+            local_address = line.group("gateway") + "/128"
             ret.append(
                 {
                     "addr_family": "inet6",
                     "destination": local_address,
                     "gateway": "::",
                     "netmask": "",
                     "flags": "U",
-                    "interface": ip_interface,
+                    "interface": ip_interface if ip_interface else "",
                 }
             )
         else:
-            address_mask = convert_cidr(comps[0])
-            ip_interface = ""
-            if comps[1] == "dev":
-                ip_interface = comps[2]
+            ip_interface = line.group("interface")
+            ip_gateway = line.group("gateway")
+            if line.group("gateway") is None:
+                flags = "U"
+            else:
+                flags = "UG"
 
             ret.append(
                 {
                     "addr_family": "inet6",
-                    "destination": comps[0],
-                    "gateway": "::",
+                    "destination": line.group("network"),
+                    "gateway": ip_gateway if ip_gateway else "::",
```

Comment:
**suggestion**: another place where the ternary could just be `ip_gateway or "::"`